supabase/functions/alert-data-processing/index.ts

```diff
@@ -26,164 +26,109 @@ interface EmailConfig {
-                if (evaluation_interval.includes(":")) {
-                    const timeOffset = evaluation_interval.split(":");
-                    const hours = Number(timeOffset[0]);
-
-                    // Ideally, an hour-based interval less than ten would be represented by a single digit. To accomplish this,
-                    // the hour segment of the evaluation interval is selected (i.e., timeOffset[0]) and attempted to be converted to a number.
-                    // This conditional is a failsafe, in the event the aforementioned conversion fails which would result in the display
-                    // of two digits for the hour (e.g., 02 hours instead of 2 hours).
-                    formattedEvaluationInterval = isFinite(hours) ? `${hours} hours` : `${timeOffset[0]} hours`;
-                }
-
-                const subject = `Estuary Flow: Alert for ${spec_type} ${catalog_name}`;
-
-                const detailsPageURL = getTaskDetailsPageURL(catalog_name, spec_type);
-
-                const content =
-                    `<p>You are receiving this alert because your task, ${spec_type} ${catalog_name} hasn't seen new data in ${formattedEvaluationInterval}.  You can locate your task <a href="${detailsPageURL}" target="_blank" rel="noopener">here</a> to make changes or update its alerting settings.</p>`;
-
-                return {
-                    content,
-                    emails,
-                    subject,
-                };
-            },
-        )
-        : [];
-
-    const pendingConfirmationEmails: EmailConfig[] = confirmations
-        ? confirmations.map(
-            ({ arguments: { emails, spec_type }, catalog_name }) => {
-                const subject = `Estuary Flow: Alert for ${spec_type} ${catalog_name}`;
-
-                const detailsPageURL = getTaskDetailsPageURL(catalog_name, spec_type);
-
-                const content =
-                    `<p>You are receiving this notice because a previous alert for your task, ${spec_type} ${catalog_name}, has now resolved.  You can locate your task <a href="${detailsPageURL}" target="_blank" rel="noopener">here</a> to make changes or update its alerting settings.</p>`;
-
-                return {
-                    content,
-                    emails,
-                    subject,
-                };
-            },
-        )
-        : [];
-
-    const pendingEmails = [...pendingAlertEmails, ...pendingConfirmationEmails];
-
-    if (pendingEmails.length === 0) {
-        return new Response(null, {
-            headers: { ...corsHeaders, "Content-Type": "application/json" },
-            status: 200,
-        });
-    }
+    const pendingEmail: EmailConfig =
+        alertRecord.resolved_at === null
+            ? formatAlertEmail(alertRecord)
+            : formatConfirmationEmail(alertRecord);
 
     const resendToken = Deno.env.get('RESEND_API_KEY');
```

Comment:
We do need an additional authorization check in this edge function, because it can be used to send (limited) email content to arbitrary email addresses.

There are many ways to do this, and I'm pretty ambivalent on exactly how, but I'll give a suggested approach to make it concrete:
 * Add an additional passed-in Deno.env `SHARED_SECRET`, which corresponds to a `alert-email-fn-shared-secret` secret I just created within the database using the Supabase vault extension: https://supabase.com/dashboard/project/eyrcnmuzzyriypdajwdk/settings/vault/secrets
 * When building requests, the value can then be accessed via view `vault.decrypted_secrets` and passed as an `Authorization` header.
 * Replace the existing public API key Authorization header with this ☝️ by using the `--no-verify-jwt` flag as discussed below.

Reply:
Another option, which might be a bit simpler overall: We can add the Resend API key as a vaulted secret in supabase, and pass it into the edge function.

This makes the edge function itself unprivileged: it's only useful if you already have a valid Resend API key to call it with.

Reply:
I have not been able to get `supabase/vault` working locally. I am running into the same issue reported in this [discussion](https://github.com/orgs/supabase/discussions/9022) when I attempt to create the `supabase_vault` extension (ref: [`supabase/vault` installation instructions](https://github.com/supabase/vault#installation)).

Error reported by the trigger function when attempting to fetch the secret:
```
ERROR:  relation "vault.decrypted_secrets" does not exist
LINE 1: select decrypted_secret            from vault.decrypted_secr...
```

Error reported when attempting to create the extension:
```
Error running SQL: could not open extension control file "/usr/share/postgresql/14/extension/supabase_vault.control": No such file or directory
```

Reply:
Taking a look myself. If we can't get it working, I think it'd be fine to use a regular table to store & plumb through the credential -- it just can't be in our migrations. `vault` looked nice and easy with built-in UI so :shrug:

Reply:
**NOTE:** Updating the major version of PostgreSQL used by Supabase to 15 enabled support for the `supabase_vault` extension. Johnny made a handful of small tweaks to the following migrations that will be rolled up into this PR: _10_specs_ext.sql_ and _11_stats.sql_. The details of this note will be called out in the PR description explicitly for general awareness.